### PR TITLE
Added make and Delete connections routes

### PR DESCRIPTION
**Pull request by calebrs**
Alright, lots of new code. Probably bugs as well, but it's functional. Here's the scoop:

- Added the `connectionController` which has two routes.
- The POST route does two things: verifies the connection is good, and then adds it to the database if it is.
- The DELETE route is self-explanatory.
- I created `connection.js` in the db folder which contains all the queries for connections, including a test method which I used to test the `eventDbQuery` method.
- I created the `event-db-query.js` file, which has two methods. One that verifies the connection, one that allows users to query the connected database.
- Edited the README with the new endpoints.

NOTE: Security is low at this point, and should probably go into the backlog.

### CODE REVIEW

File: server/db/verifyConnection.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const { Client } = require("pg");
+const { dbServerQuery } = require("./db-query");
+
+// send an initial query to the db server and make sure that it does not throw an error
+module.exports = {
+  async verifyConnection({ user, host, password, database, port }) {
```
Comment (julia-martin) [inline server/db/verifyConnection.js]: Is this file needed if there's already the same function in the `event-db-query.js` file? I don't see it being used anywhere so far

File: server/db/db-query.js [hunk 2]
```diff
@@ -2,23 +2,25 @@ const { Client } = require("pg");
 require("dotenv").config();
 
 const logQuery = (statement, parameters) => {
-  let timeStamp = new Date();
-  let formattedTimeStamp = timeStamp.toString().substring(4, 24);
+  const timeStamp = new Date();
+  const formattedTimeStamp = timeStamp.toString().substring(4, 24);
   console.log(formattedTimeStamp, statement, parameters);
```
Comment (julia-martin) [inline server/db/db-query.js:7]: I'm curious what this `logQuery` function's purpose is since there's no return value, Is it just for debugging?

File: server/db/event-db-query.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      host,
+      password,
+      database,
+      port,
+    });
+
+    await client.connect();
+
+    console.log(
+      `Connection to host: ${host} database: ${database} successfully verified.`
```
Comment (julia-martin) [inline server/db/event-db-query.js:46]: How do we know if the connection is unsuccessful? What's the behavior of an unsuccessful `connect()`?

**Review by julia-martin — COMMENTED** [review]
Wow this is really impressive, thanks for doing all this! My main questions were on some duplicated code I saw (`verifyConnection.js`), and testing the connection for failures. But other than that looks great!

File: server/db/verifyConnection.js (hunk 1 above)
Comment (calebrs) [reply server/db/verifyConnection.js]: Ah, good catch! I forgot to delete it while I was experimenting with different options.

File: server/db/db-query.js (hunk 2 above)
Comment (calebrs) [reply server/db/db-query.js:7]: It's purely for logging/debugging. It's a relic of the "todo" app code I copied this from.

File: server/db/event-db-query.js (hunk 3 above)
Comment (calebrs) [reply server/db/event-db-query.js:46]: Yeah, this took me a while to figure out and I'm not 100% sure my solution is the best option.

Firstly, it's assumed that all the error handling for the methods in `db-query` and `event-Db-Query` are handled in the route where they're called. If you look at the routes Sean created, this is the case. 

So the error handling in this case is contained in the `connectionController` file when the query is called. But, like you said, how do we differentiate the errors? In the catch block on lines 22-27 of `connectionController` I ended up accessing a property in the error object to check what kind of error it is. That's what determines what message is sent back to the client.

Again, not 100% sure if this is the best way to do it but it worked well when I tested it.

**Comment by seanrichardson95** [comment]
Great work here Caleb! I'm looking forward to seeing this in action during stand-up today. Once we get consensus on what we return from "/api/connection" (per my comment above), let's merge!

File: server/db/event-db-query.js (hunk 3 above)
Comment (julia-martin) [reply server/db/event-db-query.js:46]: I see, I think I'm still confused about this exact part since there's no query being run here, so based on my understanding those error handlers won't be reached. I was wondering if we should maybe add a SELECT query between `await client.connect()` and the success message?

Comment (calebrs) [reply server/db/event-db-query.js:46]: When `client.connect()` on line 43 of `event-db-query` is run, either an error will be thrown, or it will be successful. I thought about trying to do a query here, but an error would be thrown before I could even query anything.
